### ISSUE TITLE
Android Crash: com.google.android.exoplayer2.util.Assertions.checkArgument - java.lang.IllegalArgumentException

### ISSUE THREAD

**Issue by mitchdowney** [podverse/podverse-rn#1714]
This is our most common crash on Android right now.

```
com.google.android.exoplayer2.util.Assertions.checkArgument
Assertions.java, line 39
java.lang.IllegalArgumentException
...
com.google.android.exoplayer2.util.Assertions.checkArgument Assertions.java:39
com.google.android.exoplayer2.ExoPlayerImpl.removeMediaItemsInternal ExoPlayerImpl.java:2189
com.google.android.exoplayer2.ExoPlayerImpl.removeMediaItems ExoPlayerImpl.java:655
com.google.android.exoplayer2.BasePlayer.removeMediaItem BasePlayer.java:79
com.doublesymmetry.kotlinaudio.players.QueuedAudioPlayer.remove QueuedAudioPlayer.kt:155
com.doublesymmetry.kotlinaudio.players.QueuedAudioPlayer.remove QueuedAudioPlayer.kt:168
com.doublesymmetry.trackplayer.service.MusicService.remove MusicService.kt:276
com.doublesymmetry.trackplayer.module.MusicModule$remove$1.invokeSuspend MusicModule.kt:317
kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith ContinuationImpl.kt:33
kotlinx.coroutines.DispatchedTask.run DispatchedTask.kt:106
android.os.Handler.handleCallback Handler.java:942
android.os.Handler.dispatchMessage Handler.java:99
android.os.Looper.loopOnce Looper.java:226
android.os.Looper.loop Looper.java:313
android.app.ActivityThread.main ActivityThread.java:8747
java.lang.reflect.Method.invoke Method.java
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run RuntimeInit.java:571
com.android.internal.os.ZygoteInit.main ZygoteInit.java:1067
```

**Comment by mitchdowney** [comment]
This should hopefully be fixed (or at least avoided with a workaround) in v4.12.5.

https://github.com/podverse/podverse-rn/pull/1715/files

**Comment by mitchdowney** [comment]
Hopefully this KotlinAudio PR by puckey will fix this issue: https://github.com/doublesymmetry/react-native-track-player/issues/1953

**Comment by mitchdowney** [comment]
A bug report that sounds related:

> I'm using Moto G-Power, with Android 11. Within the last two weeks, my podverse app has been full of problems. I have plenty of space for storage of media, and my phone is working fine.

> I can use the download feature and download a podcast, with no problem (I don't stream anything). When I press the play Button the app will shut down and close. After launching again, the episode will not play. Sometimes I can delete the episode and download it again, and it may, or may not play.

> This happens with all podcasts now.

**Comment by mitchdowney** [comment]
This appears to have been fixed in 4.12.7 🥳 our crash and ANR rate has dropped by 20% since 4.12.6.
